From 8e90c68fc0a7d5bce035fe7939eb03fa5cf333a2 Mon Sep 17 00:00:00 2001
From: Christine Caulfield <ccaulfie@redhat.com>
Date: Thu, 25 Jun 2015 11:18:58 +0100
Subject: [PATCH] totemip: Be more selective when matching bindnetaddr to IP addresses

For instance, the IPs are:
eth0: 10.150.10.81/8
eth1: 10.150.10.91/24

The bindnetaddr in corosync is: 10.150.10.0 and the addr of eth1 is
selected in UDPU mode. Then the bindto address mistakenly
choosees the addr of eth0.

This does not affect binding to a fully-specific IP address.

Signed-off-by: Lidong Zhong <lzhong@suse.com>
Reviewed-by: Christine Caulfield <ccaulfie@redhat.com>
---
 exec/totemip.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/exec/totemip.c b/exec/totemip.c
index cc8e11f..012a944 100644
--- a/exec/totemip.c
+++ b/exec/totemip.c
@@ -626,7 +626,7 @@ int totemip_iface_check(struct totem_ip_address *bindnet,
 					memcpy(&network, RTA_DATA(tb[IFA_LOCAL]), sizeof(uint32_t));
 					memcpy(&addr, bindnet->addr, sizeof(uint32_t));
 
-					if ((addr & netmask) == (network & netmask)) {
+					if ((addr) == (network & netmask)) {
 						memcpy(ipaddr.addr, RTA_DATA(tb[IFA_ADDRESS]), TOTEMIP_ADDRLEN);
 						found_if = 1;
 					}
-- 
1.7.1

